reference/normalize.go

```diff
@@ -1,9 +1,124 @@
 package reference
 
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution/digest"
+)
+
 var (
-	defaultTag = "latest"
+	legacyDefaultDomain = "index.docker.io"
+	defaultDomain       = "docker.io"
+	defaultRepoPrefix   = "library/"
+	defaultTag          = "latest"
```

Comment:
I would prefer not to, ideally they are only used through functions and hidden from the callers. The places the default tag are used today should rather be using `EnsureTagged`. Other use is for logging, but the "default" tag is irrelevant to the logs, the useful information is that a reference was altered along with before/after. 

Only other types used are here https://github.com/docker/docker/blob/master/registry/config.go#L258. That function will likely get replaced to use the parsers in this package to validate.